src/main/java/com/google/cloud/dns/ZoneInfo.java

```diff
@@ -320,6 +327,22 @@ public String getNonExistence() {
     public String getState() {
       return state;
     }
+
+    private static void validateState(String state) {
+      List<String> states = Arrays.asList("on", "off", "transfer");
+      if (!states.contains(state)) {
+        throw new IllegalArgumentException(
+            "Invalid state value, Acceptable values are 'on', 'off' or 'transfer' only");
+      }
+    }
+
+    private static void validateNonExistence(String nonExistence) {
+      List<String> values = Arrays.asList("nsec", "nsec3");
+      if (!values.contains(nonExistence)) {
+        throw new IllegalArgumentException(
+            "Invalid NonExistence value, Acceptable values are 'nsec', or 'nsec3' only");
+      }
+    }
```

Comment:
it's not good to create arraylist of valid values each time the method is called.
I would suggest:
```
private static final Set<String> validStateValues = ...
private static final Set<String> validNonExistanceValues = ...

private static void validateValue(String value, Set<String>validValues) {
 ...
}
```

Reply:
done. @dmitry-fa PTAL

Reply:
Looks good! Thanks